mesoscope/cellpack
# staging: run recipes from firebase

Problem
=======
What is the problem this work solves, including
closes #167 

Solution
========
This is a staging branch for run recipes from firebase.

features in this branch that need to be reviewed:
- a new DBUploader class for data conversion and uploading recipes into modules

already reviewed: 
- firebase credential input
- AWSHandler for storing simularium results to S3
- save metadata to firebase 
- open a new tab for simularium result 
- handlers are initiated using the DATABASE_IDS enum

## Type of change
Please delete options that are not relevant.

* New feature (non-breaking change which adds functionality)

Steps to Verify:
----------------
1. Have firebase project setup and your personal credentials ready
2. Upload a recipe to your firebase: `upload -r examples/recipes/v2/gradients.json`
3. Pack the remote recipe you uploaded: `pack -r firebase:recipes/gradients_v_default -c examples/packing-configs/run.json`
4. check up the result on Simularium, AWS S3 and Firebase

Keyfiles (delete if not relevant):
-----------------------
1. `DBRecipeHandler.py`
2. `autopack/__init__.py`
